### ISSUE TITLE
Print verbose output (-v) on failures

### ISSUE THREAD

**Issue by bep** [frankban/quicktest#144]
quicktest's output is much nicer when running with the `-v` flag, but it's not a great experience having that flag on all the time, and it's often a little fiddly to toggle it on and off in editors.

So, I would be happy if quicktest could make the verbose output the default on `qt.Assert` failures.

I was reminded about this by @rogpeppe 's fix here https://github.com/rogpeppe/go-internal/pull/148 -- before that I always ran with the `-v` flag when using the `testscripts` package, which was annoying in general, because I only needed the verbose output when it failed.

**Comment by rogpeppe** [comment]
Can you say specifically which output you're talking about please? An example would be helpful.

**Comment by bep** [comment]
@rogpeppe a simple example below.

* Printing the full got/want is often much easier to reason about than the diff (especially true for tests you haven't looked at in a while)
* It also allows you to more or less copy/paste the got into want in the test.

```go
func TestFoo(t *testing.T) {
	c := qt.New(t)
	c.Assert([]string{"a", "b", "c", "d"}, qt.DeepEquals, []string{"a", "b", "c"})
}

```

```
--- FAIL: TestFoo (0.00s)
go test -run TestFoo     
slicehelpers_test.go:70: 
    error:
      values are not deep equal
    diff (-got +want):
        []string{
            "a",
            "b",
            "c",
      -     "d",
        }
    stack:
      /Users/bep/dev/go/bep/helpers/slicehelpers/slicehelpers_test.go:70
        c.Assert([]string{"a", "b", "c", "d"}, qt.DeepEquals, []string{"a", "b", "c"})
```

```
go test -v -run TestFoo     
slicehelpers_test.go:70: 
  error:
    values are not deep equal
  diff (-got +want):
      []string{
          "a",
          "b",
          "c",
    -     "d",
      }
  got:
    []string{"a", "b", "c", "d"}
  want:
    []string{"a", "b", "c"}
  stack:
    /Users/bep/dev/go/bep/helpers/slicehelpers/slicehelpers_test.go:70
      c.Assert([]string{"a", "b", "c", "d"}, qt.DeepEquals, []string{"a", "b", "c"})
```

**Comment by frankban** [comment]
This behavior was introduced as a fix for https://github.com/frankban/quicktest/issues/33
Probably we can investigate being a bit smarter about cases in which verbose should be required.
Single line values really do not increase the noise.

**Comment by frankban** [comment]
https://github.com/frankban/quicktest/pull/147 is an attempt to fix this. That's released as `v1.14.4`. Quicktest now only suppress the got/want args if they are longer than ten lines (and -v is not provided). We can adjust this gradually as required, but it should address the cases you provided, in which the output for args is one line. Closing this ticket, but please reopen it if the fix is not good enough!
